synapseclient/client.py

```diff
@@ -1354,6 +1355,115 @@ def uploadFileHandle(self, path, parent, synapseStore=True, mimetype=None, md5=N
         """
         return upload_file_handle(self, parent, path, synapseStore, md5, file_size, mimetype)
 
+    ############################################################
+    #                  Download List                           #
+    ############################################################
+    def clear_download_list(self):
+        """Clear all files from download list"""
+        self.restDELETE("/download/list")
+
+    def remove_from_download_list(self, list_of_files: typing.List[typing.Dict]) -> int:
+        """Remove a batch of files from download list
+
+        :param: array of files in the format of a mapping
+                {fileEntityId: synid, versionNumber: version}
+
+        :returns: Number of files removed from download list
+        """
+        request_body = {"batchToRemove": list_of_files}
+        num_files_removed = self.restPOST(
+            "/download/list/remove",
+            body=json.dumps(request_body)
+        )
+        return num_files_removed
```

Comment:
What if the call to restPOST fails?

Reply:
the restPOST will return the respective errors